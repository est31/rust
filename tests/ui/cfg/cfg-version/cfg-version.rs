#[cfg(version(42))] //~ ERROR: expected a version literal
fn foo() {}
#[cfg(version(1.20))] //~ ERROR: expected a version literal
fn foo() -> bool { true }
#[cfg(version = "1.20")] //~ WARN: unexpected `cfg` condition name: `version`
fn foo() -> bool { true }
#[cfg(version("1.44"))]
fn foo() -> bool { true }
#[cfg(not(version("1.44")))]
fn foo() -> bool { false }

#[cfg(version("1.43", "1.44", "1.45"))] //~ ERROR: expected single version literal
fn bar() -> bool  { false }
#[cfg(version(false))] //~ ERROR: expected a version literal
fn bar() -> bool  { false }
#[cfg(version("foo"))] //~ WARNING: unknown version literal format
fn bar() -> bool  { false }
#[cfg(version("999"))] //~ WARNING: unknown version literal format
fn bar() -> bool  { false }
#[cfg(version("-1"))] //~ WARNING: unknown version literal format
fn bar() -> bool  { false }
#[cfg(version("65536"))] //~ WARNING: unknown version literal format
fn bar() -> bool  { false }
#[cfg(version("0"))] //~ WARNING: unknown version literal format
fn bar() -> bool { true }
#[cfg(version("1.0"))]
fn bar() -> bool { true }
#[cfg(version("1.65536.2"))] //~ WARNING: unknown version literal format
fn bar() -> bool  { false }
#[cfg(version("1.20.0-stable"))] //~ WARNING: unknown version literal format
fn bar() {}

fn main() {
    assert!(foo());
    assert!(bar());
    assert!(cfg!(version("1.42")));
}
