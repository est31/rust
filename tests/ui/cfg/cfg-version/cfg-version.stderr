error: expected a version literal
  --> $DIR/cfg-version.rs:1:15
   |
LL | #[cfg(version(42))]
   |               ^^

error: expected a version literal
  --> $DIR/cfg-version.rs:3:15
   |
LL | #[cfg(version(1.20))]
   |               ^^^^

error: expected single version literal
  --> $DIR/cfg-version.rs:12:7
   |
LL | #[cfg(version("1.43", "1.44", "1.45"))]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected a version literal
  --> $DIR/cfg-version.rs:14:15
   |
LL | #[cfg(version(false))]
   |               ^^^^^

warning: unknown version literal format, assuming it refers to a future version
  --> $DIR/cfg-version.rs:16:15
   |
LL | #[cfg(version("foo"))]
   |               ^^^^^

warning: unknown version literal format, assuming it refers to a future version
  --> $DIR/cfg-version.rs:18:15
   |
LL | #[cfg(version("999"))]
   |               ^^^^^

warning: unknown version literal format, assuming it refers to a future version
  --> $DIR/cfg-version.rs:20:15
   |
LL | #[cfg(version("-1"))]
   |               ^^^^

warning: unknown version literal format, assuming it refers to a future version
  --> $DIR/cfg-version.rs:22:15
   |
LL | #[cfg(version("65536"))]
   |               ^^^^^^^

warning: unknown version literal format, assuming it refers to a future version
  --> $DIR/cfg-version.rs:24:15
   |
LL | #[cfg(version("0"))]
   |               ^^^

warning: unknown version literal format, assuming it refers to a future version
  --> $DIR/cfg-version.rs:28:15
   |
LL | #[cfg(version("1.65536.2"))]
   |               ^^^^^^^^^^^

warning: unknown version literal format, assuming it refers to a future version
  --> $DIR/cfg-version.rs:30:15
   |
LL | #[cfg(version("1.20.0-stable"))]
   |               ^^^^^^^^^^^^^^^

warning: unexpected `cfg` condition name: `version`
  --> $DIR/cfg-version.rs:5:7
   |
LL | #[cfg(version = "1.20")]
   |       ^^^^^^^^^^^^^^^^
   |
   = help: expected names are: `FALSE` and `test` and 32 more
   = help: to expect this configuration use `--check-cfg=cfg(version, values("1.20"))`
   = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg.html> for more information about checking conditional configuration
   = note: `#[warn(unexpected_cfgs)]` on by default

error: aborting due to 4 previous errors; 8 warnings emitted

